server_environment/serv_config.py

```diff
@@ -97,9 +97,20 @@ def _load_config():
         config_p.read(conf_files)
     except Exception as e:
         raise Exception('Cannot read config files "%s":  %s' % (conf_files, e))
+
     config_p.read(system_base_config.rcfile)
     config_p.remove_section('options')
 
+    env_config = os.getenv('SERVER_ENV_CONFIG')
+    if env_config:
+        try:
+            config_p.read_string(env_config)
+        except configparser.Error as err:
+            raise Exception(
+                'SERVER_ENV_CONFIG content could not be parsed: %s'
+                % (err,)
```

Comment:
Have you checked `err` does not reveal the content of the of the environment variable?

Reply:
Good catch, it does!

```
2018-07-19 06:18:57,343 1 CRITICAL odoodb odoo.modules.module: Couldn't load module server_environment
2018-07-19 06:18:57,343 1 CRITICAL odoodb odoo.modules.module: SERVER_ENV_CONFIG content could not be parsed: File contains no section headers.
file: '<string>', line: 1
'my super secret but invalid format'
2018-07-19 06:18:57,345 1 WARNING odoodb odoo.modules.loading: Transient module states were reset
2018-07-19 06:18:57,345 1 ERROR odoodb odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/opt/odoo/external-src/server-env/server_environment/serv_config.py", line 107, in _load_config
    config_p.read_string(env_config)
  File "/usr/lib/python3.5/configparser.py", line 720, in read_string
    self.read_file(sfile, source)
  File "/usr/lib/python3.5/configparser.py", line 715, in read_file
    self._read(f, source)
  File "/usr/lib/python3.5/configparser.py", line 1077, in _read
    raise MissingSectionHeaderError(fpname, lineno, line)
configparser.MissingSectionHeaderError: File contains no section headers.
file: '<string>', line: 1
'my super secret but invalid format'
```

I'll see how to prevent that.